src/aero_rt/src/lib.rs

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2021 The Aero Project Developers.
+ *
+ * This file is part of The Aero Project.
+ *
+ * Aero is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Aero is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Aero. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#![feature(lang_items, start)]
+#![no_std]
+
+use aero_syscall::*;
+
+#[no_mangle]
+unsafe extern "C" fn _start(argc: isize, argv: *const *const u8) -> ! {
+    extern "C" {
+        fn main(_: isize, _: *const *const u8) -> isize;
+    }
+
+    sys_open("/dev/tty", OpenFlags::O_RDONLY).expect("Failed to open stdin");
+    sys_open("/dev/tty", OpenFlags::O_WRONLY).expect("Failed to open stdout");
+    sys_open("/dev/tty", OpenFlags::O_WRONLY).expect("Failed to open stderr");
```

Comment:
We do not need to manually open `stdin`, `stdout` and `stderr` on every start function and only once in the init binary which we currently do in the shell itself since its the first process that runs. In future we might have a separate init binary for this since the init process will fork it self resulting in forking the file descriptors too. So maybe move this to the shell's main function?

Reply:
Well I'd assume that child processes wouldn't inherit parent's file descriptors? Surely each process needs to have them opened once again, but in case you `fork` this code won't be ran, unless you `exec` afterwards which *should* result in a fresh set of file descriptors, otherwise having opened a file like `/etc/passwd` as read/write could give unwanted write access to a critical system file to a potentially unprivileged process. I'm not entirely familiar with UNIX/POSIX so please correct me if I'm wrong 😄

Reply:
Okay I guess this line of code in the `Task::fork` code answers my question, let me fix that quickly!
https://github.com/Andy-Python-Programmer/aero/blob/7d6a9063806c5d27c96343e9fb2042e02cd17509/src/aero_kernel/src/userland/task.rs#L171

Reply:
> but in case you fork this code won't be ran, unless you exec afterwards which should result in a fresh set of file descriptors, otherwise having opened a file like /etc/passwd as read/write could give unwanted write access to a critical system file to a potentially unprivileged process.

You close the file descriptors after use :^) So if you close `/etc/passwd` before fork everything is fine. Also remember it just creates a copy of the parent's fds not a link to em.

Reply:
Fixed this in 9c3cfc3378d0884ed5975919040bd4365d4113c9!